File: java/word-count/src/main/java/WordCount.java
import java.util.Arrays;
import java.util.Map;
import java.util.HashMap;

public class WordCount {
    public Map<String, Integer> phrase(String input) {
        Map<String, Integer> result = new HashMap<>();

        Arrays.stream(input.toLowerCase().split("\\W+"))
              .forEach(word ->
                  result.put(word, result.getOrDefault(word, 0) + 1)
              );

        return result;
    }
}

File: python/word-count/word_count.py
import re
from functools import reduce

def word_count(phrase):
    return reduce(lambda count_map, word: count_map.update({ word: count_map.get(word, 0) + 1}) or count_map, re.findall(r"[a-z0-9]+(?:'[a-z0-9]+)?", phrase.lower()), {})
